iron-spi-aws/src/main/java/io/axway/iron/spi/aws/AwsUtils.java

```diff
@@ -36,29 +45,54 @@ public static void setAws(AwsClientBuilder builder, //
      * @param action the action to retry
      * @param retryLimit retry number limit
      * @param durationInMillis duration between each retry
-     * @throws LimitExceededException after retry exhausted
+     * @throws AwsException after retry exhausted
      */
     public static <T> T performAmazonActionWithRetry(String actionLabel, Supplier<T> action, int retryLimit, int durationInMillis) {
         int retryCount = 0;
         do {
             try {
                 return action.get();
-            } catch (LimitExceededException lee) {
+            } catch (LimitExceededException | ProvisionedThroughputExceededException | KMSThrottlingException e) {
                 // We should just wait a little time before trying again
                 int remainingRetries = retryLimit - retryCount;
-                LOG.debug("LimitExceededException caught", args -> args.add("action", actionLabel).add("remainingRetryCount", remainingRetries));
-            }
-            try {
-                LOG.debug("Throttling", args -> args.add("action", actionLabel).add("durationMs", durationInMillis));
-                Thread.sleep(durationInMillis);
-            } catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
-                throw new AwsException("Thread interrupted while performing action", args -> args.add("action", actionLabel), e);
+                LOG.debug("Amazon exception caught",
+                          args -> args.add("exception", e.getClass().getName()).add("action", actionLabel).add("remainingRetryCount", remainingRetries));
```

Comment:
why not using the normal way to report exception though alf, ie 3rd parameter is for exception